File: OnlineStore/Api/CommonEntities/Product.cs
using CommonEntities.Additional;

namespace CommonEntities
{
	public class Product : BaseDbEntity
	{

		public decimal Price { get; set; }
		//public string Status { get; set; }
		public int IdProductInformation { get; set; }

		public ProductStatus ProductStatus { get; set; }
		public ProductInformation ProductInformation { get; set; }

		#region Конструктор

		public Product()
		{

		}

		public Product(int id, decimal price, string status, int idProductInformation) : base(id)
		{
			Price = price;
			
			IdProductInformation = idProductInformation;
			ProductStatus = new ProductStatus(status);
			ProductInformation = null;
		}
		public Product(decimal price, string status, int idProductInformation)
		{
			Price = price;
			IdProductInformation = idProductInformation;
			ProductStatus = new ProductStatus(status);
			ProductInformation = null;
		}


		public Product(decimal price, ProductStatus status, ProductInformation productInformation)
		{
			Price = price;
			//Status = status.GetStatusName();
			ProductInformation = productInformation;
			if (productInformation?.IdEntity != null)
				IdProductInformation = productInformation.IdEntity.Value;
			ProductStatus = status;
		}

		#endregion

	}
}

File: OnlineStore/OnlineStoreUI/src/model/entities/apiEntities/product.ts
import { ProductInformation } from './productInformation';
import { ProductStatus } from './additional/productStatus';

export class Product{
    idEntity: number;
    price:number;
    idProductInformation:number;
    productStatus:ProductStatus;
    productInformation: ProductInformation; 


    constructor(){
        this.idEntity = 0;
        this.price = 0;
        this.idProductInformation = 0;
        this.productStatus = null;
        this.productInformation = new ProductInformation();
    }
}